src/managers/ThreadManager.js

```diff
@@ -0,0 +1,232 @@
+   * @param {ThreadChannelResolvable|FetchThreadsOptions} [options] If a ThreadChannelResolvable, the thread to fetch.
+   * If undefined, fetches all active threads.
+   * If an Object, fetches the specified threads.
+   * @param {BaseFetchOptions} [cacheOptions] Additional options for this fetch
+   * only applies when fetching a single thread
+   * @returns {Promise<?(ThreadChannel|FetchedThreads)>}
+   * @example
+   * // Fetch a thread by its id
+   * channel.threads.fetch('831955138126104859')
+   *   .then(channel => console.log(channel.name))
+   *   .catch(console.error);
+   */
+  fetch(options, { cache = true, force = false } = {}) {
+    if (!options) return this.fetchActive(cache);
+    const channel = this.client.channels.resolveID(options);
+    if (channel) return this.client.channels.fetch(channel, cache, force);
+    if (options.archived) {
+      return this.fetchArchived(options.archived, cache);
+    }
+    return this.fetchActive(cache);
+  }
+
+  /**
+   * Data that can be resolved to give a Date object. This can be:
+   * * A Date object
+   * * A number representing a timestamp
+   * * An ISO8601 string
+   * @typedef {Date|number|string} DateResolvable
+   */
+
+  /**
+   * The options used to fetch archived threads.
+   * @typedef {Object} FetchArchivedThreadOptions
+   * @property {string} [type='public'] The type of threads to fetch, either `public` or `private`
+   * @property {boolean} [fetchAll=false] When type is `private` whether to fetch **all** archived threads,
+   * requires `MANAGE_THREADS` if true
+   * @property {DateResolvable|ThreadChannelResolvable} [before] Identifier for a Date or Snowflake
+   * to get threads that were created before it,
+   * must be a ThreadChannelResolvable when type is `private` and fetchAll is `false`
+   * @property {number} [limit] Maximum number of threads to return
+   */
+
+  /**
+   * The data returned from a thread fetch that returns multiple threads.
+   * @typedef {FetchedThreads}
+   * @property {Collection<Snowflake, ThreadChannel>} threads The threads fetched, with any members returned
+   * @property {?boolean} hasMore Whether there are potentially additional threads that require a subsequent call
+   */
+
+  /**
+   * Obtains a set of archived threads from Discord, requires `READ_MESSAGE_HISTORY` in the parent channel.
+   * @param {FetchArchivedThreadOptions} [options] The options to use when fetch archived threads
+   * @param {boolean} [cache=true] Whether to cache the new thread objects if they aren't already
+   * @returns {Promise<FetchedThreads>}
+   */
+  async fetchArchived({ type = 'public', fetchAll = false, before, limit } = {}, cache = true) {
+    let path = this.client.api.channels(this.channel.id);
+    if (type === 'private' && fetchAll) {
+      path = path.users('@me');
+    }
+    let timestamp;
+    let id;
+    if (typeof before !== 'undefined') {
+      if (before instanceof ThreadChannel || /[0-9]{16,19}/g.test(String(before))) {
```

Comment:
Should probably make the regex stricter?
```suggestion
      if (before instanceof ThreadChannel || /^\d{16,19}$/.test(String(before))) {
```

Reply:
yes yes yes